addOns/postman/src/main/java/org/zaproxy/addon/postman/models/Request.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.postman.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import java.util.List;
+import org.zaproxy.addon.postman.deserializers.ListDeserializer;
+import org.zaproxy.addon.postman.deserializers.ObjectDeserializer;
+
+/**
+ * Represents the request in the Postman format which is contained by the items.
+ *
+ * @see https://learning.postman.com/collection-format/reference/request/
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Request {
+    private static final List<String> ALLOWED_METHODS =
+            List.of(
+                    "GET",
+                    "PUT",
+                    "POST",
+                    "PATCH",
+                    "DELETE",
+                    "COPY",
+                    "HEAD",
+                    "OPTIONS",
+                    "LINK",
+                    "UNLINK",
+                    "PURGE",
+                    "LOCK",
+                    "UNLOCK",
+                    "PROPFIND",
+                    "VIEW");
+
+    @JsonDeserialize(using = ObjectDeserializer.class)
+    private Url url;
+
+    @JsonDeserialize(using = ObjectDeserializer.class)
+    private String method;
+
+    @JsonDeserialize(using = ListDeserializer.class)
+    private List<KeyValueData> header;
+
+    @JsonDeserialize(using = ObjectDeserializer.class)
+    private Body body;
+
+    public Request() {
+        this.method = "GET";
+    }
+
+    public Request(String url) {
+        this.method = "GET";
```

Comment:
Should this just be done at declaration? (like line 57 above)

Also there's probably a core constant for most methods/verbs.